torchaudio/datasets/ljspeech.py

```diff
@@ -0,0 +1,79 @@
+import os
+
+import torchaudio
+from torchaudio.datasets.utils import download_url, extract_archive, walk_files
+from torch.utils.data import Dataset
+
+URL = "https://data.keithito.com/data/speech/LJSpeech-1.1.tar.bz2"
+FOLDER_IN_ARCHIVE = "wavs"
+
+
+class LJSPEECH(Dataset):
+    """
+    Create a Dataset for LJSpeech-1.1. Each item is a tuple of the form:
+    waveform, sample_rate, transcript, normalized_transcript
+    """
+
+    _ext_audio = ".wav"
+    _ext_archive = '.tar.bz2'
+
+    def __init__(
+            self, root, url=URL, folder_in_archive=FOLDER_IN_ARCHIVE, download=False
+    ):
+
+        basename = os.path.basename(url)
+        archive = os.path.join(root, basename)
+
+        basename = basename.split(self._ext_archive)[0]
+        folder_in_archive = os.path.join(basename, folder_in_archive)
+
+        self._path = os.path.join(root, folder_in_archive)
+        self._metadata_path = os.path.join(root, basename, 'metadata.csv')
+
+        if download:
+            if not os.path.isdir(self._path):
+                if not os.path.isfile(archive):
+                    download_url(url, root)
+                extract_archive(archive)
+
+        walker = walk_files(
+            self._path, suffix=self._ext_audio, prefix=False, remove_suffix=True
+        )
+        self._walker = list(walker)
+        self._metadata = self._load_metadata()
+
+    def _load_metadata(self):
```

Comment:
We can avoid creating a variable just for the metadata by following the example done in [CommonVoice](https://github.com/pytorch/audio/blob/master/torchaudio/datasets/commonvoice.py#L107). Essentially, in this case, we iterate over the rows of the file to generate the walker. The load item function then just opens the audio file and returns the desired tuple.